### PR TITLE
TST/CLN: Split out some to_string tests

### PR DESCRIPTION
**Pull request by dsaxton**
test_format.py is over 3000 lines long so breaking some (but not all) to_string-specific tests into a separate test_to_string.py file (we already have test_to_html.py, test_to_latex.py, and test_to_markdown.py)

### CODE REVIEW

File: pandas/tests/io/formats/test_to_string.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+def test_repr_embedded_ndarray():
+    arr = np.empty(10, dtype=[("err", object)])
+    for i in range(len(arr)):
+        arr["err"][i] = np.random.randn(i)
+
+    df = DataFrame(arr)
+    repr(df["err"])
+    repr(df)
+    df.to_string()
```
Comment (dsaxton) [inline pandas/tests/io/formats/test_to_string.py:18]: Are we missing an assertion here (same below)?

Comment (jreback) [reply pandas/tests/io/formats/test_to_string.py:18]: sure, but do as a followup

**Review by jreback — APPROVED** [review]
looks fine. can you confirm we have same number of tests before & after here.

**Comment by dsaxton** [comment]
> looks fine. can you confirm we have same number of tests before & after here.

master
```
pandas/tests/io/formats/test_format.py .............................................................................................................................................................. [ 67%]
.............................................................................                                                                                                                         [100%]

============================================================================================ 235 passed in 4.39s ============================================================================================
```

branch
```
pandas/tests/io/formats/test_format.py .............................................................................................................................................................. [ 78%]
...........................................                                                                                                                                                           [100%]

============================================================================================ 201 passed in 4.21s ============================================================================================
```

```
pandas/tests/io/formats/test_to_string.py ..................................                                                                                                                          [100%]

============================================================================================ 34 passed in 0.42s =============================================================================================
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton

**Pull request merged**